src/Adapter/Ftp.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Bolt\Filesystem\Adapter;
+
+use League\Flysystem\Adapter\Ftp as FtpAdapter;
+use League\Flysystem\Config;
+use Symfony\Component\Debug\Exception\ContextErrorException;
+
+/**
+ * FTP adapter.
+ *
+ * @author Gawain Lynch <gawain.lynch@gmail.com>
+ */
+class Ftp extends FtpAdapter
+{
+    /** @var int */
+    protected $filePerm = 0664;
+    /** @var int */
+    protected $directoryPerm = 0775;
+
+    /**
+     * @inheritdoc
+     */
+    public function __construct(array $config)
+    {
+        $this->configurable[] ='directoryPerm';
+        parent::__construct($config);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function has($path)
+    {
+        $location = $this->applyPathPrefix($path);
+
+        return (bool) $this->getMetadata($location);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function createDir($dirname, Config $config)
+    {
+        $location = $this->applyPathPrefix($dirname);
+        if ($this->has($location)) {
+            // For mirror() to work
+            return ['path' => $dirname, 'type' => 'dir'];
+        }
+
+        return parent::createDir($dirname, $config);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function createActualDirectory($directory, $connection)
+    {
+        try {
+            $result = parent::createActualDirectory($directory, $connection);
+        } catch (ContextErrorException $e) {
```

Comment:
This is only thrown if Symfony's Error handler is hooked up. If you are trying to silence errors set a temporary error handler or use `@`.